### PR TITLE
Tanner/qt6 not type errors

### PR DESCRIPTION
**Pull request by tanneryould**
# Description

Resolves errors related to component type errors in QML samples. Please test these samples on Windows and see how they compare to the 100.15 sample viewer release.

## Type of change

- [x] Bug fix
- [ ] New sample implementation
- [ ] Sample viewer enhancement
- [ ] Other enhancement

**Platforms tested on**:

- [ ] Windows
- [ ] Android
- [ ] Linux
- [x] macOS
- [ ] iOS

## Checklist

- [ ] Runs and compiles on all active platforms as a standalone sample
- [ ] Runs and compiles in the sample viewer(s)
- [ ] Branch is up to date with the latest main/v.next
- [ ] All merge conflicts have been resolved
- [ ] Self-review of changes
- [ ] There are no warnings related to changes
- [ ] No unrelated changes have been made to any other code or project files
- [ ] Code is commented with correct formatting (CTRL+i)
- [ ] All variable and method names are camel case
- [ ] There is no leftover commented code
- [ ] Screenshots are correct size and display in description tab (500px by 500px, platform agnostic)
- [ ] If adding a new sample, it is added to the sample viewer
- [ ] Cherry-picked to Main branch (if applicable)

### CODE REVIEW

File: ArcGISRuntimeSDKQt_QMLSamples/EditData/EditFeaturesWithFeatureLinkedAnnotation/EditFeaturesWithFeatureLinkedAnnotation.qml [hunk 1]
```diff
@@ -177,13 +176,6 @@ Rectangle {
         radius: 10
         visible: false
 
-        GaussianBlur {
-            anchors.fill: updateWindow
-            source: mapView
-            radius: 40
-            samples: 20
-        }
-
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/EditData/EditFeaturesWithFeatureLinkedAnnotation/EditFeaturesWithFeatureLinkedAnnotation.qml:186]: This sample gets stuck on a white blank screen

File: ArcGISRuntimeSDKQt_QMLSamples/Features/FeatureLayer_Query/FeatureLayer_Query.qml [hunk 2]
```diff
@@ -101,7 +101,7 @@ Rectangle {
                 placeholderText: "Enter a state name to select"
                 inputMethodHints: Qt.ImhNoPredictiveText
                 selectByMouse: true
-                validator: RegExpValidator{ regExp: /^[a-zA-Z ]*$/ }
+                validator: RegularExpressionValidator{ regularExpression: /^[a-zA-Z ]*$/ }
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/Features/FeatureLayer_Query/FeatureLayer_Query.qml:104]: This one also gets stuck on a blank white screen

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/BrowseOGCAPIFeatureService/BrowseOGCAPIFeatureService.qml [hunk 3]
```diff
@@ -111,9 +112,7 @@ Rectangle {
         // Pop-up error message box
         MessageDialog {
             id: errorMessageBox
-            title: "Error message!"
             text: errorMessage
-            icon: StandardIcon.Warning
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/BrowseOGCAPIFeatureService/BrowseOGCAPIFeatureService.qml:116]: Same here, I wonder if it is related to:

```
qrc:/Samples/Layers/BrowseOGCAPIFeatureService/BrowseOGCAPIFeatureService.qml:104: TypeError: Cannot read property 'loadStatus' of null
```

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayDimensions/DisplayDimensions.qml [hunk 4]
```diff
@@ -86,9 +87,7 @@ Rectangle {
         // Pop-up error message box
         MessageDialog {
             id: errorMessageBox
-            title: "Error message!"
             text: errorMessage
-            icon: StandardIcon.Warning
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayDimensions/DisplayDimensions.qml:91]: Same issue with the blank screen in this sample as well

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.qml [hunk 5]
```diff
@@ -172,8 +172,8 @@ Rectangle {
                     width: 200
                     text: ""
                     selectByMouse: true
-                    validator: RegExpValidator {
-                        regExp: /(0[1-9]|1[012])[/](0[1-9]|[12][0-9]|3[01])[/](19|20)\d\d/
+                    validator: RegularExpressionValidator {
+                        regularExpression: /(0[1-9]|1[012])[/](0[1-9]|[12][0-9]|3[01])[/](19|20)\d\d/
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.qml:176]: Blank screen here too

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/RasterLayerFile/RasterLayerFile.qml [hunk 6]
```diff
@@ -92,7 +92,7 @@ Rectangle {
         folder: dataPath
         supportedExtensions: supportedFormats
 
-        onRasterFileChosen: {
+        onRasterFileChosen: url => {
             createAndAddRasterLayer(url);
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/RasterLayerFile/RasterLayerFile.qml:96]: Fails to build on Windows due to `#include <QApplication>`
```
D:\applications\qt\sdk\samples\samplecode\arcgis-runtime-samples-qt\ArcGISRuntimeSDKQt_QMLSamples\Layers\RasterLayerFile\main.cpp:15: error: C1083: Cannot open include file: 'QApplication': No such file or directory
```

File: ArcGISRuntimeSDKQt_QMLSamples/Routing/RouteAroundBarriers/RouteAroundBarriers.qml [hunk 7]
```diff
@@ -43,7 +44,6 @@ Rectangle {
 
     MessageDialog {
         id: messageDialog
-        title: "Route around barriers"
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/Routing/RouteAroundBarriers/RouteAroundBarriers.qml:46]: Blank window. Crashes as soon as I interact with the application

File: ArcGISRuntimeSDKQt_QMLSamples/Search/SearchDictionarySymbolStyle/SearchDictionarySymbolStyle.qml [hunk 8]
```diff
@@ -100,7 +100,7 @@ Rectangle {
                             Layout.fillWidth: true
                             placeholderText: repeaterModel[index] +" (e.g. "+ hintsModel[index] +")"
                             selectByMouse: true
-                            validator: RegExpValidator{ regExp: /^\s*[\da-zA-Z_][\da-zA-Z\s_]*$/ }
+                            validator: RegularExpressionValidator{ regularExpression: /^\s*[\da-zA-Z_][\da-zA-Z\s_]*$/ }
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/Search/SearchDictionarySymbolStyle/SearchDictionarySymbolStyle.qml:103]: Blank window. Crashes as soon as I interact with the application

File: ArcGISRuntimeSDKQt_QMLSamples/UtilityNetwork/PerformValveIsolationTrace/PerformValveIsolationTrace.qml [hunk 9]
```diff
@@ -442,7 +443,6 @@ Rectangle {
 
     MessageDialog {
         id: messageDialog
-        title: "Perform valve isolation trace"
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/UtilityNetwork/PerformValveIsolationTrace/PerformValveIsolationTrace.qml:445]: Blank screen, crashes on interaction

**Review by Gela — CHANGES_REQUESTED** [review]
QML samples do not render any maps, they just show a blank white screen and crash as soon as I interact with them

**Comment by tanneryould** [comment]
@Gela the discrepancies between MacOS and Windows are wild! I'll swap over to working on my Windows machine to implement and test these changes. Thank you!

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/BrowseOGCAPIFeatureService/BrowseOGCAPIFeatureService.qml (hunk 3 above)
Comment (tanneryould) [reply ArcGISRuntimeSDKQt_QMLSamples/Layers/BrowseOGCAPIFeatureService/BrowseOGCAPIFeatureService.qml:116]: Are you referring to `TypeError: Cannot read property 'loadStatus' of null`? That's a non-breaking error that just indicates that featureLayer was null when the button was trying to resolve whether or not it should be enabled. However featureLayer wasn't instantiated yet so it couldn't resolve featureLayer.loadStatus. I'll make the following change that enables the load button if the featureCollectionListComboBox.model has items. 

```js
-                    enabled: featureLayer.loadStatus !== Enums.LoadStatusLoading
+                    enabled: featureCollectionListComboBox.model.length > 0
```

**Review by Gela — APPROVED** [review]

**Pull request merged**